frontend/src/components/cluster-dashboard/cluster-list-view/ClusterListView.tsx

```diff
@@ -0,0 +1,158 @@
+  const [confDialogContent, setConfDialogContent] = React.useState("");
+  const [confDialogData, setConfDialogData] = React.useState({});
+  const [addNodeAddress, setAddNodeAddress] = React.useState("");
+  const clusterData: Cluster = props.clusterData;
+  const akkaGetClusterMembersUrl = akkaClusterProps["akka.management.url"] + "/cluster/members";
+
+  const handleMember = (member: string, mode: string): void => {
+    setConfDialogTitle(mode.toLowerCase() === "leave" ? "Remove Node From Cluster?" : "Shutdown Node in Cluster?");
+    setConfDialogContent(
+      mode.toLowerCase() === "leave" ? 
+      "Are you sure you want to remove node " + member + " from the cluster?" : 
+      "Are you sure you want to shutdown node " + member + " in the cluster?"
+    );
+    setConfDialogData({ member: member, mode: mode });
+    setOpenConfDialog(true);
+  }
+
+  const addMember = (): void => {
+    if(!addNodeAddress || addNodeAddress.trim() === "") { return; }
+    
+    axios.post(akkaGetClusterMembersUrl, 
+        querystring.stringify({ address: addNodeAddress.trim() }),
+        { headers: { "Content-Type": "application/x-www-form-urlencoded" }}
+      )
+      .then(response => {
+        if(response.status === 200) {
+          setSnackBarMessage(response.data.message);
+          setOpenSnackBar(true);
+          props.refreshClusterData();
+        }
+      })
+      .catch(function (error) {
+        console.log(error);
+        setSnackBarMessage(error.message && error.message.message ? error.message.message : error.message);
+        setOpenSnackBar(true);
+      })
+  }
+
+  const handleMemberConfirm = ({ member, mode }: { member: string; mode: string }): void => {
+    setOpenConfDialog(false);
+    axios.put(akkaGetClusterMembersUrl + "/" + member.split('://')[1], 
+        querystring.stringify({ operation: mode }),
+        { headers: { "Content-Type": "application/x-www-form-urlencoded" }}
+      )
+      .then(response => {
+        if(response.status === 200) {
+          setSnackBarMessage(response.data.message);
+          setOpenSnackBar(true);
+          props.refreshClusterData();
+        }
+      })
+      .catch(function (error) {
+        console.log(error);
+        setSnackBarMessage(error.message && error.message.message ? error.message.message : error.message);
+        setOpenSnackBar(true);
+      })
+  }
+
+  return (
+    <React.Fragment>
+      <Grid item xs={3} className="shukra-left-container pos-rel">
+        <div className="member-list-container">
+          <div className="member-list-container-title">
+            CLUSTER NODES
+          </div>
+          <TreeView className="member-tree-view" defaultCollapseIcon={<ArrowDropDownIcon />} defaultExpandIcon={<ArrowRightIcon />}>
+          {
+            clusterData && clusterData.members ? clusterData.members.map((member, index) => {
+              return <TreeItem key={index} nodeId={String(index)} label={
```

Comment:
This might be throwing a warning, but you should avoid using array indices as `key`s in a map. 
https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318

Reply:
TIL

Reply:
Changed this to use "nodeUid" field, which is a unique id.